crates/modelardb_storage/src/lib.rs

```diff
@@ -311,13 +321,264 @@ fn apache_parquet_writer_properties(
+
+    let type_array = modelardb_types::array!(record_batch, 0, StringArray);
+    let name_array = modelardb_types::array!(record_batch, 1, StringArray);
+    let schema_array = modelardb_types::array!(record_batch, 2, BinaryArray);
+    let error_bounds_array = modelardb_types::array!(record_batch, 3, ListArray);
+    let generated_columns_array = modelardb_types::array!(record_batch, 4, ListArray);
+
+    for row_index in 0..record_batch.num_rows() {
+        let table_type = type_array.value(row_index);
+        let table_name = name_array.value(row_index).to_owned();
+        let schema = try_convert_bytes_to_schema(schema_array.value(row_index).to_vec())?;
+
+        match table_type {
+            "normal" => normal_table_metadata.push((table_name, schema)),
+            "model" => {
+                let error_bounds = array_to_error_bounds(error_bounds_array.value(row_index))?;
+
+                let generated_columns = array_to_generated_columns(
+                    generated_columns_array.value(row_index),
+                    &schema.clone().to_dfschema()?,
+                )?;
+
+                let metadata = ModelTableMetadata::try_new(
+                    table_name,
+                    Arc::new(schema),
+                    error_bounds,
+                    generated_columns,
+                )?;
+
+                model_table_metadata.push(metadata);
+            }
+            _ => {
+                return Err(ModelarDbStorageError::InvalidArgument(format!(
+                    "Table type '{table_type}' is not supported.",
+                )));
+            }
+        }
+    }
+
+    Ok((normal_table_metadata, model_table_metadata))
+}
+
+/// Parse the error bound values in `error_bounds_array` into a list of [`ErrorBounds`](ErrorBound).
+/// Returns [`ModelarDbServerError`] if an error bound value is invalid.
+fn array_to_error_bounds(error_bounds_array: ArrayRef) -> Result<Vec<ErrorBound>> {
+    // unwrap() is safe since error bound values are always f32.
+    let value_array = error_bounds_array
+        .as_any()
+        .downcast_ref::<Float32Array>()
+        .unwrap();
+
+    let mut error_bounds = Vec::with_capacity(value_array.len());
+    for value in value_array.iter().flatten() {
+        if value < 0.0 {
+            error_bounds.push(ErrorBound::try_new_relative(-value)?);
+        } else {
+            error_bounds.push(ErrorBound::try_new_absolute(value)?);
+        }
+    }
+
+    Ok(error_bounds)
+}
+
+/// Parse the generated column expressions in `generated_columns_array` into a list of optional
+/// [`GeneratedColumns`](GeneratedColumn). Returns [`ModelarDbServerError`] if a generated column
+/// expression is invalid.
+fn array_to_generated_columns(
+    generated_columns_array: ArrayRef,
+    df_schema: &DFSchema,
+) -> Result<Vec<Option<GeneratedColumn>>> {
+    // unwrap() is safe since generated column expressions are always strings.
+    let expr_array = generated_columns_array
```

Comment:
Maybe use the `array!()` macro to shorten the code and make it easy to return the error in the future for casts by changing the macro.

Reply:
`array!()` is used to extract a column from a record batch which is not the case here. Since this pattern is only used a few times in this file, I have not added a new macro for this specific case.